components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/IdentifierHandler.java

```diff
@@ -408,7 +421,20 @@ protected void processAuthenticationResponse(HttpServletRequest request,
         authProperties.put("username", username);
 
         persistUsername(context, username);
-        context.setSubject(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(username));
+
+        AuthenticatedUser user;
+        if (userId != null) {
+            // If the ValidateUsername is set to true, user id will be populated and we can create the user object
+            // based on that.
+            user = new AuthenticatedUser();
+            user.setUserId(userId);
+            user.setTenantDomain(tenantDomain);
+            user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(username));
+            user.setUserName(tenantAwareUsername);
+        } else {
+            user = AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(username);
```

Comment:
This method will try to resolve the username. So this can be problematic since the ValidateUsername says not to validate the user.
@darshanasbg, @pulasthi7 any thoughts?